lib/ansible/playbook/role/__init__.py

```diff
@@ -222,7 +223,45 @@ def _load_role_data(self, role_include, parent_role=None):
         else:
             self._metadata = RoleMetadata()
 
+        # The argument_specs file will contain a dict, and can have multiple keys,
+        # included a 'main' item. But set one up explicitly if the yaml doesnt provide
+        # one. Could also do things like 'pick the first one' or try to match role name or
+        # 'from_tasks' filename to argument_spec keys, etc
+        argument_specs = {'main': {}}
+        try:
+            argument_specs = self._load_role_yaml('meta', main='argument_specs')
+        except AnsibleParserError as e:
+            pass
+
+        # TODO: need a playbook.base.Base derived object here?
+        # TODO: do we want a Role (or Task or Base) object to have a arg_spec attribute?
+        #       Seems like it would be handy for introspection and error reporting and copy()
+        self._argument_specs = argument_specs
```

Comment:
makes sense to hang it on the role, since that is what we are checking in this case